File: PMMP/src/jkorn/pvpcore/commands/PvPAreaCommand.php
<?php
/**
 * Created by PhpStorm.
 * User: jkorn2324
 * Date: 2020-06-26
 * Time: 18:57
 */

namespace jkorn\pvpcore\commands;


use jkorn\pvpcore\player\PvPCPlayer;
use jkorn\pvpcore\utils\Utils;
use pocketmine\command\Command;
use pocketmine\command\CommandSender;
use pocketmine\command\utils\CommandException;
use pocketmine\Player;
use pocketmine\utils\TextFormat;

class PvPAreaCommand extends Command
{

    public function __construct()
    {
        parent::__construct("pvparea", "Sets the pvparea information in the player.", "Usage: /pvparea <pos1:pos2>", ["pvpareapos"]);
        parent::setPermission("pvpcore.permission.edit");
    }

    /**
     * @param CommandSender $sender
     * @param string $commandLabel
     * @param string[] $args
     *
     * @return mixed
     * @throws CommandException
     */
    public function execute(CommandSender $sender, string $commandLabel, array $args)
    {
        if (!$this->testPermission($sender)) {
            return true;
        }

        if (!$sender instanceof PvPCPlayer) {
            if ($sender instanceof Player) {
                $message = TextFormat::RED . "Internal Plugin Error. Please leave and join back to use this command.";
            } else {
                $message = TextFormat::RED . "Console can't use this command.";
            }
            $sender->sendMessage(Utils::getPrefix() . " " . $message);
            return true;
        }

        if (count($args) <= 0 || (($firstArg = $args[0]) !== "pos1" && $firstArg !== "pos2")) {
            $sender->sendMessage(Utils::getPrefix() . " " . TextFormat::RED . $this->getUsage());
            return true;
        }


        if ($firstArg === "pos1") {
            $sender->setFirstPos();
        } else {
            $sender->setSecondPos();
        }

        return true;
    }
}

File: Nukkit/src/pvpcore/commands/PvPAreaCommand.java
package pvpcore.commands;

import cn.nukkit.Player;
import cn.nukkit.command.Command;
import cn.nukkit.command.CommandSender;
import cn.nukkit.utils.TextFormat;
import pvpcore.player.PvPCorePlayer;
import pvpcore.utils.Utils;

public class PvPAreaCommand extends Command {

    public PvPAreaCommand()
    {
        super("pvparea", "Sets the pvparea information in the player", "Usage: /pvparea <pos1:pos2>", new String[]{"pvpareapos"});
        super.setPermission("pvpcore.permission.edit");
    }

    /**
     * Executes the command.
     * @param commandSender - The command sender.
     * @param s - The command label.
     * @param strings - The arguments for the commands.
     * @return true if command executes successfully, false otherwise.
     */
    @Override
    public boolean execute(CommandSender commandSender, String s, String[] strings)
    {
        if(!this.testPermission(commandSender))
        {
            return true;
        }

        if(!(commandSender instanceof PvPCorePlayer))
        {
            String message;
            if(commandSender instanceof Player)
            {
                message = TextFormat.RED.toString() + "Internal Plugin Error. Please leave and join back to use this command.";
            } else {
                message = TextFormat.RED.toString() + "Console can't use this command.";
            }
            commandSender.sendMessage(Utils.getPrefix() + " " + message);
            return true;
        }

        String firstArg;
        if(strings.length <= 0 || (!(firstArg = strings[0]).equals("pos1") && !firstArg.equals("pos2")))
        {
            commandSender.sendMessage(Utils.getPrefix() + " " + TextFormat.RED.toString() + this.getUsage());
            return true;
        }

        if(firstArg.equals("pos1"))
        {
            ((PvPCorePlayer) commandSender).setFirstPos();
        } else {
            ((PvPCorePlayer) commandSender).setSecondPos();
        }

        return true;
    }
}
